### DISCUSSION TITLE
How do I trigger a full rebuild?

### DISCUSSION THREAD

**Discussion by diehard2** [microsoft/vcpkg#27357] [General]
I've been creating a custom package and my current method of triggering a rebuild is to delete the cache and delete the vcpkg submodule (or clean it) and then re-add it. There must be a better way,  but the internet isn't helping me at all. 

Also, if I change the triplet to a custom triplet and the package is installed a rebuild is not triggered. It just says my package with the original triplet is installed and doesn't do anything else. The command I'm using is

vcpkg/vcpkg install statsd --overlay-triplets=custom-triplets --overlay-ports=vcpkg-registry/ports/statsd 

If I uninstall and the install again it picks up the custom triplet, but does not rebuild any of the dependencies with the new triplet. Is this intended behavior?

**Comment by Neumann-A** [comment]
You are missing --triplet=yourtriplet or did you set env VCPKG_DEFAULT_TRIPLET?

**Comment by diehard2** [comment]
I was using the procedure outlined here - https://github.com/microsoft/vcpkg/blob/master/docs/examples/overlay-triplets-linux-dynamic.md

**Comment by diehard2** [comment]
Thanks, specifying the triplet with the overlay was what was needed. Out of curiousity, is there a flag to trigger a full rebuild without delete vcpkg or cleaning it with git?

**Reply by Neumann-A** [reply]
make sure you are also using `--host-triplet=yourtriplet` if you are not crosscompiling.

> Out of curiousity, is there a flag to trigger a full rebuild without delete vcpkg or cleaning it with git?

Use manifest mode? That will automatically check the hashes of everything a rebuild as necessary. If you want a full rebuild just delete the /installed/ folder
